Doc/library/warnings.rst

```diff
@@ -522,6 +522,52 @@ Available Functions
    and calls to :func:`simplefilter`.
 
 
+.. decorator:: deprecated(msg, *, category=DeprecationWarning, stacklevel=1)
+
+   Decorator to indicate that a class, function or overload is deprecated.
+
+   Usage::
+
+      from warnings import deprecated
+      from typing import overload
+
+      @deprecated("Use B instead")
+      class A:
+          pass
+
+      @deprecated("Use g instead")
+      def f():
+          pass
+
+      @overload
+      @deprecated("int support is deprecated")
+      def g(x: int) -> int: ...
+      @overload
+      def g(x: str) -> int: ...
+
+   When this decorator is applied to an object,
+   :term:`static type checkers <static type checker>`
+   will generate a diagnostic on usage of the deprecated object.
```

Comment:
I'd put this paragraph below what's currently the second paragraph

Reply:
Done; also applied your other suggestions. Thanks for the review!